mlir/lib/Dialect/Transform/IR/TransformOps.cpp

```diff
@@ -866,6 +870,7 @@ matchBlock(Block &block, Operation *op, transform::TransformState &state,
   // Remember the values mapped to the terminator operands so we can
   // forward them to the action.
   ValueRange yieldedValues = block.getTerminator()->getOperands();
+  mappings.clear();
```

Comment:
I am not sure I understand why this is now necessary, does this merit a comment?